cloud-service-providers/aws/eks/nim-eks-cdk/lib/eks-cluster-stack.ts

```diff
@@ -0,0 +1,140 @@
+import * as cdk from "aws-cdk-lib";
+import { Construct } from "constructs";
+import {
+  aws_eks as eks,
+  aws_ec2 as ec2,
+  aws_iam as iam,
+} from "aws-cdk-lib";
+import { AlbControllerVersion, Cluster } from "aws-cdk-lib/aws-eks";
+import { Vpc } from "aws-cdk-lib/aws-ec2/lib/vpc";
+import { KubectlV29Layer } from "@aws-cdk/lambda-layer-kubectl-v29";
+import { Peer, Port } from "aws-cdk-lib/aws-ec2";
+
+interface EksClusterStackProps extends cdk.StackProps {
+  vpc: Vpc;
+}
+
+export class EksClusterStack extends cdk.Stack {
+  readonly cluster: Cluster;
+
+  constructor(scope: Construct, id: string, props: EksClusterStackProps) {
+    super(scope, id, props);
+
+    // Define IAM policy statement to allow list access to eks cluster
+    const eksPolicyStatement = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ["eks:*"],
+      resources: ["*"],
+    });
```

Comment:
I'm not as familiar with TypeScript style guides, but it looks like there should be a newline in a few places in these two typescript files for better readability.

Reply:
Added new line.